packages/embark-accounts-manager/src/index.ts

```diff
@@ -0,0 +1,117 @@
+  private nodeAccounts: string[];
+  private web3: any;
+  private ready: boolean;
+
+  constructor(private readonly embark: Embark, _options: any) {
+    this.logger = embark.logger;
+    this.events = embark.events;
+    this.accounts = [];
+    this.nodeAccounts = [];
+    this.ready = false;
+
+    this.events.setCommandHandler("accounts-manager:onReady", (cb) => {
+      if (this.ready) {
+        return cb();
+      }
+      this.events.once("accounts-manager:ready", cb);
+    });
+    this.events.request("proxy:onReady", () => {
+      this.parseAccounts();
+    });
+
+    this.events.setCommandHandler("accounts:get", (cb: any) => {
+      cb(null, this.accounts);
+    });
+  }
+
+  private async checkBlockchainRequest(params: any, callback: (error: any, result: any) => void) {
+    if (params.reqData.method === "eth_sendTransaction" && this.accounts.length) {
+      // Check if we have that account in our wallet
+      const account = this.accounts.find((acc) => Web3.utils.toChecksumAddress(acc.address) === Web3.utils.toChecksumAddress(params.reqData.params[0].from));
+      if (account && account.privateKey) {
+        return this.web3.eth.accounts.signTransaction(params.reqData.params[0], account.privateKey , (err: any, result: any) => {
+          if (err) {
+            return callback(err, null);
+          }
+          params.reqData.method = "eth_sendRawTransaction";
+          params.reqData.params = [result.rawTransaction];
+          callback(err, params);
+        });
+      }
+    }
+    callback(null, params);
+  }
+
+  public arrayEqual(arrayA: string[], arrayB: string[]) {
+    if (arrayA.length !== arrayB.length) {
+      return false;
+    } else {
+      return arrayA.every((address, index) => Web3.utils.toChecksumAddress(address) === Web3.utils.toChecksumAddress(arrayB[index]));
+    }
+  }
+
+  private async checkBlockchainResponse(params: any, callback: (error: any, result: any) => void) {
+    if ((params.reqData.method === "eth_accounts" || params.reqData.method === "personal_listAccounts") && this.accounts.length) {
+      if (!this.arrayEqual(params.respData.result, this.nodeAccounts)) {
+        this.nodeAccounts = params.respData.result;
+        this.accounts = AccountParser.parseAccountsConfig(this.embark.config.blockchainConfig.accounts, this.web3, dappPath(), this.logger, this.nodeAccounts);
+      }
+      params.respData.result = this.accounts.map((acc) => {
+        if (acc.address) {
+          return acc.address;
+        }
+        return acc;
+      });
+      return callback(null, params);
+    }
+    callback(null, params);
+  }
+
+  private setReady() {
+    if (this.ready) {
+      return;
+    }
+    this.embark.registerActionForEvent("blockchain:proxy:request", this.checkBlockchainRequest.bind(this));
+    this.embark.registerActionForEvent("blockchain:proxy:response", this.checkBlockchainResponse.bind(this));
+    this.ready = true;
+    this.events.emit("accounts-manager:ready");
+  }
```

Comment:
I agree about the proxy being a Stack component, but I still need the `ready` on accounts-manager because I can't modify the transactions or the getAccounts until the accounts have been parsed and funded.

Reply:
OK looks good, but could we please remove the entire `setReady` function altogether, in favour of simply setting `this.ready = true;` on line 163? The `accounts-manager:ready` event is not being used.